### ISSUE TITLE
U-Turns on trunk and between trunk_link and trunk

### ISSUE THREAD

**Issue by Aiurek** [osmandapp/Osmand#7388]
Lenovo Moto G4 plus, Android 7.0
OsmAnd+ 3.4.xx, before the play store release 2019-08-03
May style: OsmAnd
Navigation service: OsmAnd (offline)

Being tired I almost followed the car routing direction illustrated in the below image, which would have sent me in the complete opposite direction on the highway. Encountered min. 3 cases in the last month, were Osmand told me to enter the highway in the wrong direction (luckily not as a wrong-way driver) and do a U-turn as soon as i exit the acceleration lane (which is prohibited and not possible). 
Wouldn't it be possible to generally block such u turns by setting a threshold for possible turns on highway=trunk and between highway=trunk_link and highway=trunk?

https://www.openstreetmap.org/#map=17/48.90580/8.65298

![Bild11](https://user-images.githubusercontent.com/27865200/62708003-c0d63a00-b9f2-11e9-9934-2451080a4d80.jpg)

**Comment by scaidermern** [comment]
Traffic signal penalties? By taking this weird route OsmAnd avoids two traffic signals ([these two, at the start and at the end](https://www.openstreetmap.org/directions?engine=fossgis_osrm_car&route=48.90616%2C8.65117%3B48.90496%2C8.65313#map=19/48.90557/8.65215)).

Apart from that, is there a u-turn restriction missing in OSM?

**Comment by Aiurek** [comment]
Yes, the traffic signals might really contribute to that behaviour. And for sure a missing turn restrictions, whereby there are to many missing in OSM.
What about defining the maximum angle for guidance on trunks or between trunk links and trunks to be e.g. 80°?

**Comment by vshcherb** [comment]
That's not possible to block u-turns cause they are needed in many routes
